dashboard/app/models/unit.rb

```diff
@@ -1598,6 +1598,29 @@ def unit_without_lesson_plans?
     lessons.none?(&:has_lesson_plan)
   end
 
+  def summarize_for_lesson_materials_view(user)
+    summary = {
+      id: id,
+      title: title_for_display,
+      name: name,
+      link: script_path(self),
+      unitNumber: unit_number,
+      version_year: version_year,
+      course_id: unit_group.try(:id),
+      courseVersionId: get_course_version&.id,
+      courseOfferingId: get_course_version&.course_offering&.id,
+      scriptOverviewPdfUrl: get_unit_overview_pdf_url,
+      scriptResourcesPdfUrl: get_unit_resources_pdf_url,
+      teacher_resources: resources.sort_by(&:name).map(&:summarize_for_resources_dropdown),
+      student_resources: student_resources.sort_by(&:name).map(&:summarize_for_resources_dropdown),
+    }
+    # Only get lessons with lesson plans
+    filtered_lessons = lessons.select(&:has_lesson_plan)
+    summary[:lessons] = filtered_lessons.map {|lesson| lesson.summarize_for_lesson_materials(user)}
```

Comment:
Again, let's keep the API response as slim as possible here.

Reply:
Yep yep - this one there was a lot more of "eh, I don't know, some day we might use some of these..." thinking going on.  I removed anything where I could not concretely say "Yes. This will be used in the next 7 calendar days for planned tasks."